src/main/java/net/imglib2/roi/geom/integer/Line.java

```diff
@@ -0,0 +1,415 @@
+		long maxDiff = -1;
+		for ( int d = 0; d < n; d++ )
+		{
+			final long dx = end.getLongPosition( d ) - start.getLongPosition( d );
+			diff.setPosition( dx, d );
+			if ( Math.abs( dx ) > maxDiff )
+				maxDiff = Math.abs( dx );
+		}
+		this.nPoints = maxDiff;
+	}
+
+	@Override
+	public boolean equals( final Object o )
+	{
+		if ( o == this ) { return true; }
+		if ( !( o instanceof Line ) ) { return false; }
+		final Line ol = ( Line ) o;
+		return locationsEqual( start, ol.start ) && locationsEqual( end, ol.end );
+	}
+
+	@Override
+	public int hashCode()
+	{
+		int hash = 17;
+		hash = 31 * hash + hash( start );
+		hash = 31 * hash + hash( end );
+		return hash;
+	}
+
+	@Override
+	public long size()
+	{
+		return nPoints;
+	}
+
+	@Override
+	public Void firstElement()
+	{
+		return cursor().next();
+	}
+
+	@Override
+	public Object iterationOrder()
+	{
+		return this;
+	}
+
+	@Override
+	public double realMin( final int d )
+	{
+		return min( d );
+	}
+
+	@Override
+	public void realMin( final double[] min )
+	{
+		for ( int d = 0; d < n; d++ )
+			min[ d ] = realMin( d );
+	}
+
+	@Override
+	public void realMin( final RealPositionable min )
+	{
+		for ( int d = 0; d < n; d++ )
+			min.setPosition( realMin( d ), d );
+	}
+
+	@Override
+	public double realMax( final int d )
+	{
+		return max( d );
+	}
+
+	@Override
+	public void realMax( final double[] max )
+	{
+		for ( int d = 0; d < n; d++ )
+			max[ d ] = realMax( d );
+	}
+
+	@Override
+	public void realMax( final RealPositionable max )
+	{
+		for ( int d = 0; d < n; d++ )
+			max.setPosition( realMax( d ), d );
+	}
+
+	@Override
+	public int numDimensions()
+	{
+		return n;
+	}
+
+	@Override
+	public Iterator< Void > iterator()
+	{
+		return cursor();
+	}
+
+	@Override
+	public long min( final int d )
+	{
+		return Math.min( start.getLongPosition( d ), end.getLongPosition( d ) );
+	}
+
+	@Override
+	public void min( final long[] min )
+	{
+		for ( int d = 0; d < n; d++ )
+			min[ d ] = min( d );
+	}
+
+	@Override
+	public void min( final Positionable min )
+	{
+		for ( int d = 0; d < n; d++ )
+			min.setPosition( min( d ), d );
+	}
+
+	@Override
+	public long max( final int d )
+	{
+		return Math.max( start.getLongPosition( d ), end.getLongPosition( d ) );
+	}
+
+	@Override
+	public void max( final long[] max )
+	{
+		for ( int d = 0; d < n; d++ )
+			max[ d ] = max( d );
+	}
+
+	@Override
+	public void max( final Positionable max )
+	{
+		for ( int d = 0; d < n; d++ )
+			max.setPosition( max( d ), d );
+	}
+
+	@Override
+	public void dimensions( final long[] dimensions )
+	{
+		for ( int d = 0; d < n; d++ )
+			dimensions[ d ] = max( d ) - min( d );
+	}
+
+	@Override
+	public long dimension( final int d )
+	{
+		return max( d ) - min( d );
+	}
+
+	@Override
+	public Cursor< Void > cursor()
+	{
+		return localizingCursor();
+	}
+
+	@Override
+	public Cursor< Void > localizingCursor()
+	{
+		return new LineCursor();
+	}
+
+	private final class LineCursor implements Cursor< Void >
```

Comment:
You could extend `AbstractCursor`, but I'm not sure if it will simplify this much since you need to do the rounding.